numpy_rms/__init__.py

```diff
@@ -26,12 +24,10 @@ def rms(a: NDArray, window_size: int) -> NDArray:
         and (a.ndim == 1 or (a.ndim == 2 and a.shape[0] == 1))
         and (a.flags["C_CONTIGUOUS"] or a.flags["F_CONTIGUOUS"])
     ):
-        output_shape = list(a.shape)
-        output_shape[-1] = math.floor(a.shape[-1] / window_size)
+        output_shape = a.shape[:-2] + (a.shape[-1] // window_size,)
```

Comment:
Are you sure this is what you intended?

>>> (1,2,3)[:-2]
(1,)

At first glance, a.shape[:-1] looks more correct

Reply:
You are right :facepalm: